otherlibs/systhreads/st_posix.h

```diff
@@ -426,3 +432,48 @@ value caml_wait_signal(value sigs) /* ML */
   return Val_int(0);            /* not reached */
 #endif
 }
+
+#include "../../config/s.h"
+
+void st_determine_thread_stack_size(void** top_of_stack, size_t* stack_size)
+{
+  /* Note: at least under NPTL, the returned size will include the guard
+     page size. */
+
+  int local_var;
+
+  *top_of_stack = (void*) &local_var;
```

Comment:
I was told that an optimizing compiler of the "bug the programmer" kind could delete this assignment or assign anything to `*top_of_stack`, because it is undefined behavior for the caller of this function to use the value `*top_of_stack` (it points to an object that no longer exists).  For GCC and Clang (if GNUC is defined) you could use `__builtin_frame_address(0)`.

Also, these default assignments to `top_of_stack` and `stack_size` could (should?) come last in the function, see comment below.